src/components/Form/Form.js

```diff
@@ -97,6 +101,15 @@ export function Form({ children, config }) {
     // This just makes the rest of this easier to read
     const fieldConfig = config.fields[fieldName]
 
+    const doValidation = (field) => {
```

Comment:
Perhaps more verbose, but just moving this out made this much easier to debug so I opted to leave it like this. Functionality is the same as what Nick original had.